ext/web/16_image_data.js

```diff
@@ -0,0 +1,175 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import * as webidl from "ext:deno_webidl/00_webidl.js";
+import DOMException from "ext:deno_web/01_dom_exception.js";
+const primordials = globalThis.__bootstrap.primordials;
+const {
+  TypeError,
+} = primordials;
+
+webidl.converters["PredefinedColorSpace"] = webidl.createEnumConverter(
+  "PredefinedColorSpace",
+  [
+    "srgb",
+    "display-p3",
+  ],
+);
+class ImageData {
+  /** @type {number} */
+  #width;
+  /** @type {height} */
+  #height;
+  /** @type {Uint8Array} */
+  #data;
+  /** @type {'srgb' | 'display-p3'} */
+  #colorSpace;
+
+  constructor(arg0, arg1, arg2 = undefined, arg3 = undefined) {
+    webidl.requiredArguments(
+      arguments.length,
+      2,
+      'Failed to construct "ImageData"',
+    );
+    this[webidl.brand] = webidl.brand;
+
+    let sourceWidth;
+    let sourceHeight;
+    let data;
+    let settings;
+
+    // Overload: new ImageData(data, sw [, sh [, settings ] ])
+    if (webidl.type(arg0) === "Object") {
+      data = arg0;
+      sourceWidth = webidl.type(arg1) !== "Undefined"
+        ? parseInt(arg1, 10)
+        : undefined;
+      sourceHeight = webidl.type(arg2) !== "Undefined"
+        ? parseInt(arg2, 10)
```

Comment:
use the unsigned long converter

Reply:
I see there's also a `NumberParseInt` primordial? Would that also make sense to use?